Declare astro as peerdependency

### Describe the bug

Hi :wave:,

I was trying to add this integration to a project which uses pnpm workspaces, and hoisting is disabled.
Since `starlight-openapi` uses `astro`, it should be declared as a dependency/peerDependency, so it can be resolved from within the package.

https://github.com/HiDeoo/starlight-openapi/blob/a6e0980d5f154eb125bacdcb0aeccc136d0d34e5/packages/starlight-openapi/libs/config.ts#L1-3

Currently, when building the project I get this error:

```
  7:31:53 PM [vite] (ssr) Error when evaluating SSR module /project/astro.config.ts:
  |- Error: Cannot find module 'astro/errors' imported from '/project/node_modules/.pnpm/starlight-openapi@0.12.0@astrojs+markdown-remark@6.1.0_iro5jtkx52gzwgzqkgxf6aom74/node_modules/starlight-openapi/libs/config.ts'
      at fetchModule (file:///project/node_modules/.pnpm/vite@6.0.11_@types+node@22.13.1_jiti@2.4.2_lightningcss@1.27.0_sass@1.83.4_stylus@0.57.0_ters_3eng3dvlihtdkkkrs5g563pfbi/node_modules/vite/dist/node/chunks/dep-M1IYMR16.js:51641:19)
...
```


### To Reproduce

Create a pnpm project that uses astro, starlight and starlight-openapi (by default they will not be hoisted).
Try to run a build, and observe the error above.

### Expected behavior

`astro/errors` can be resolved properly

### How often does this bug happen?

Every time

### System Info

- operating system: macOS 15.3
- node 22.13.1
- pnpm 9.15.5
- astro 5.2.5

### Additional Context

_No response_

--- a/.changeset/itchy-kangaroos-search.md
+++ b/.changeset/itchy-kangaroos-search.md
@@ -1,0 +1,5 @@
+---
+'starlight-openapi': patch
+---
+
+Adds `astro` as a peer dependency to prevent potential build errors in monorepos with hoisting disabled.

--- a/packages/starlight-openapi/package.json
+++ b/packages/starlight-openapi/package.json
@@ -27,7 +27,8 @@
   },
   "peerDependencies": {
     "@astrojs/markdown-remark": ">=6.0.0",
-    "@astrojs/starlight": ">=0.30.0"
+    "@astrojs/starlight": ">=0.30.0",
+    "astro": ">=5.0.0"
   },
   "engines": {
     "node": ">=18.17.1"

